### PR TITLE
Add delete in mptvdb recording database when file is physically deleted in mptvseries.

### PR DESCRIPTION
**Pull request by framug**

### CODE REVIEW

**Comment by framug** [comment]
> Good to know about the order alias change, I haven't tested this change yet from @andrewjswan pull request
Well, I don't know if it will be the same with this patch on your system but on mine, I can see plenty of sql errors both in MP logs and in TVS logs, some examples :

`2019-08-22 09:39:44.322 [INFO][38]: SQL Execution Failed. Reason = 'SQLiteClient: TVSeriesDatabase4.db3 cmd:sqlite3_prepare16:pvm=null err:ERROR detailed:1st ORDER BY term does not match any column in the result set query:select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID=online_episodes.CompositeID)  where online_episodes.SeriesID > 0 and online_episodes.Hidden = 0 and online_episodes.Watched != 1 and online_episodes.CompositeID not in (select distinct local_episodes.CompositeID2 from local_episodes where local_episodes.CompositeID2 != '') union select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID2=online_episodes.CompositeID)  where online_episodes.SeriesID > 0 and online_episodes.Hidden = 0 and online_episodes.Watched != 1 and local_episodes.CompositeID2 != '' order by local_episodesFileDateAdded desc ', Query = 'select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID=online_episodes.CompositeID)  where online_episodes.SeriesID > 0 and online_episodes.Hidden = 0 and online_episodes.Watched != 1 and online_episodes.CompositeID not in (select distinct local_episodes.CompositeID2 from local_episodes where local_episodes.CompositeID2 != '') union select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID2=online_episodes.CompositeID)  where online_episodes.SeriesID > 0 and online_episodes.Hidden = 0 and online_episodes.Watched != 1 and local_episodes.CompositeID2 != '' order by local_episodesFileDateAdded desc '
2019-08-22 09:39:44.728 [INFO][38]: SQL Execution Failed. Reason = 'SQLiteClient: TVSeriesDatabase4.db3 cmd:sqlite3_prepare16:pvm=null err:ERROR detailed:1st ORDER BY term does not match any column in the result set query:select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID=online_episodes.CompositeID)  where local_episodes.EpisodeFilename != '' and local_episodes.EpisodeFilename != '' and online_episodes.Hidden = 0 and online_episodes.CompositeID not in (select distinct local_episodes.CompositeID2 from local_episodes where local_episodes.CompositeID2 != '') union select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID2=online_episodes.CompositeID)  where local_episodes.EpisodeFilename != '' and local_episodes.EpisodeFilename != '' and online_episodes.Hidden = 0 and local_episodes.CompositeID2 != '' order by online_episodesEpisodeIndex asc ', Query = 'select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID=online_episodes.CompositeID)  where local_episodes.EpisodeFilename != '' and local_episodes.EpisodeFilename != '' and online_episodes.Hidden = 0 and online_episodes.CompositeID not in (select distinct local_episodes.CompositeID2 from local_episodes where local_episodes.CompositeID2 != '') union select online_episodes.CompositeID, online_episodes.EpisodeID, online_episodes.SeriesID, online_episodes.EpisodeIndex, online_episodes.SeasonIndex, online_episodes.EpisodeName, online_episodes.Watched, online_episodes.Summary, online_episodes.FirstAired, online_episodes.OnlineDataImported, online_episodes.GuestStars, online_episodes.Director, online_episodes.Writer, online_episodes.Hidden, online_episodes.lastupdated, online_episodes.DownloadPending, online_episodes.DownloadExpectedName, online_episodes.ThumbUrl, online_episodes.thumbFilename, online_episodes.TraktSeen, online_episodes.TraktLibrary, online_episodes.follwit_id, online_episodes.PlayCount, online_episodes.LastWatchedDate, online_episodes.FirstWatchedDate, online_episodes.Rating, online_episodes.RatingCount, online_episodes.myRating, online_episodes.myRatingAt, online_episodes.Combined_episodenumber, online_episodes.Combined_season, online_episodes.DVD_chapter, online_episodes.DVD_discid, online_episodes.DVD_episodenumber, online_episodes.DVD_season, online_episodes.EpImgFlag, online_episodes.IMDB_ID, online_episodes.Language, online_episodes.ProductionCode, online_episodes.absolute_number, online_episodes.is_movie, online_episodes.seasonid, online_episodes.thumb_added, online_episodes.thumb_height, online_episodes.thumb_width, online_episodes.airsafter_season, online_episodes.airsbefore_episode, online_episodes.airsbefore_season, local_episodes.EpisodeFilename, local_episodes.OriginalComposite, local_episodes.CompositeID, local_episodes.SeriesID, local_episodes.SeasonIndex, local_episodes.EpisodeIndex, local_episodes.LocalEpisodeName, local_episodes.LocalImportProcessed, local_episodes.CompositeUpdated, local_episodes.AvailableSubtitles, local_episodes.OriginalComposite2, local_episodes.CompositeID2, local_episodes.EpisodeIndex2, local_episodes.videoWidth, local_episodes.videoHeight, local_episodes.FileDateAdded, local_episodes.FileDateCreated, local_episodes.IsAvailable, local_episodes.DateWatched, local_episodes.StopTime, local_episodes.ext, local_episodes.VolumeLabel, local_episodes.localPlaytime, local_episodes.VideoCodec, local_episodes.VideoFormat, local_episodes.VideoFormatProfile, local_episodes.VideoBitrate, local_episodes.VideoFrameRate, local_episodes.VideoAspectRatio, local_episodes.AudioCodec, local_episodes.AudioFormat, local_episodes.AudioFormatProfile, local_episodes.AudioBitrate, local_episodes.AudioChannels, local_episodes.AudioTracks, local_episodes.AudioLanguage, local_episodes.TextCount, local_episodes.FirstAired, local_episodes.VideoColourPrimaries, local_episodes.VideoFormatCommercial from online_episodes  left join local_episodes on (local_episodes.CompositeID2=online_episodes.CompositeID)  where local_episodes.EpisodeFilename != '' and local_episodes.EpisodeFilename != '' and online_episodes.Hidden = 0 and local_episodes.CompositeID2 != '' order by online_episodesEpisodeIndex asc '
...`

File: MP-TVSeries/DB Classes/DBEpisode.cs [hunk 1]
```diff
@@ -902,10 +903,23 @@ public List<string> deleteEpisode(TVSeriesPlugin.DeleteMenuItems type, bool dele
                             {
                                 // support for duplicate episodes - this will only happen from series/season deletes
                                 var files = episodes.Select(e => e[DBEpisode.cFilename].ToString()).Distinct();
+                                // for use of in MPtvDB
+                                TvBusinessLayer layer = new TvBusinessLayer();
```
Comment (damienhaynes) [inline MP-TVSeries/DB Classes/DBEpisode.cs]: Will this fail if TVServer or TVPlugin is not installed?

**Comment by framug** [comment]
Hello Damien,

Just tested, you are true, it only works in single-seat for the moment.
I modified a bit as this :

` bool MPUseTVServer = MediaPortal.Util.Utils.UsingTvServer;

if (MPUseTVServer)
{
    // Also delete recording in MPtvDB
    try
        {
            TvBusinessLayer layer = new TvBusinessLayer();
            layer.GetRecordingByFileName(f).Delete();
            MPTVSeriesLog.Write(string.Format("Also Deleting line in MP table recording"));
        }
        catch (Exception ex)
        {
            // this should succeed only when there is a record in MP database..
            MPTVSeriesLog.Write(string.Format("Seems no recording line to delete in MPtvDB : {0}", ex.Message));
        }
}
`
Therefore, it's not enough, for the moment, I have "tvbus...dll not found", even when "MPUseTVServer" is false,
I don't know why (yet).
You are the plugin owner, if you have an idea (faster than I), it would be appreciated... otherwise, I will find a way for it works in all cases but, it can take a while because of my real life... ;-)

Best regards.

Franck.

**Comment by damienhaynes** [comment]
Hey mate,

I think it's because when you enter the main method, it still sees:
TvBusinessLayer layer = new TvBusinessLayer();
layer.GetRecordingByFileName(f).Delete();


Try hiding the tv related code in another method e.g.:

bool MPUseTVServer = MediaPortal.Util.Utils.UsingTvServer;

if (MPUseTVServer)
{
// Also delete recording in MPtvDB
DeleteFromMyTvDatabase(f);
}

**Comment by framug** [comment]
Hello,
You are true, hiding TvBusinessLayer in another method did the trick.
I have tested it and commited, if you want to integrate...

**Comment by damienhaynes** [comment]
Thanks for confirming.

**Pull request merged**